### DISCUSSION TITLE
Using date comparison in a expression

### DISCUSSION THREAD

**Discussion by maverick792** [ezylang/EvalEx#390] [Q&A]
Hi,

I have to solve expressions which are a combination of string, numbers and dates. Example: a > 2 and b = "John" and date > "05-06-2010". 

Is it even possible? If not, I suggest adding code where if string can be formatted as date, then date comparison should be used. Any thoughts or suggestions?

**Comment by vincentlejeune** [comment]
Hi, 

The next 3.1 version will introduce date manipulation. You chould check how it work's here https://github.com/ezylang/EvalEx/blob/3.1.x/src/test/java/com/ezylang/evalex/functions/datetime/DateTimeFunctionsTest.java
The pull request is here https://github.com/ezylang/EvalEx/pull/381

You have to use function to create date time. 
For your example, you could write : 
`a > 2 and b = "John" and date > DT_PARSE("2010-06-05")`

**Reply by maverick792** [reply]
I suppose it would be better if the underlying library can detect that expression contains date string and parse it to date. A date format can be set when we initialize expression variable. That way the library knows the date format. 

At least that is the use case I have and I changed the evaluation value class to suit to my needs. Let me know if you would like to see it and give some feedback. I can raise a pull request.

**Reply by uklimaschewski** [reply]
Hi, sorry for the late answer.

as @vincentlejeune already noted, version 3.1 will have date and duration support included.
I am currently rearranging the conversion code in evaluation value, to make a better use of the expression configuration.

I think I will pick up your idea and add one or two additional configuration parameters, to allow automatic conversion of date, time and duration values from strings, so that instead of writing `date > DT_PARSE("2010-06-05")`, it would be possible to write `date > "2010-06-05"`.

Though I am very busy at the moment, I hope to have that all ready in a few more weeks.
